fenix/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -636,6 +637,12 @@ class HomeFragment : Fragment() {
 
         homeViewModel.sessionToDelete = null
 
+        requireComponents.appStore.state.wasLastTabClosedPrivate?.also {
+            showUndoSnackbar(requireContext().tabClosedUndoMessage(it))
+        }
+
+        requireComponents.appStore.dispatch(AppAction.TabStripAction.UpdateLastTabClosedAction(null))
+
```

Comment:
should the dispatch to reset the boolean is null only happen when the boolean is non-null?

```suggestion
        requireComponents.appStore.state.wasLastTabClosedPrivate?.also {
            showUndoSnackbar(requireContext().tabClosedUndoMessage(it))
            requireComponents.appStore.dispatch(AppAction.TabStripAction.UpdateLastTabClosedAction(null))
        }
```

Reply:
Good point, updated!